### PR TITLE
feat: navigator & animations

### PR DESCRIPTION
**Pull request by r0man1337**
Added 2 new features:
1. Navigator - you can set navigation target (hex col, row) and pointer arrow will show you moving direction and also display distance (in hexes)
2. All buildings in Hexception now play animations (if they are present in the model)

Also:
1. Added HUD Scene to game renderer to place UI HUD objects that do not dependant to changes in the zoom and rotation of the camera
2. Refactor HexagonScene code structure

### CODE REVIEW

File: client/src/three/scenes/HUDScene.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
+import * as THREE from "three";
+import { MapControls } from "three/examples/jsm/controls/MapControls";
+import { SceneManager } from "../SceneManager";
+import { Navigator } from "../components/Navigator";
+import { GUIManager } from "../helpers/GUIManager";
+// Remove import for DirectionalLightHelper
+// import { DirectionalLightHelper } from "three";
```
Comment (edisontim) [inline client/src/three/scenes/HUDScene.ts:7]: remove comments

File: client/src/three/scenes/HUDScene.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  constructor(sceneManager: SceneManager, controls: MapControls) {
+    this.scene = new THREE.Scene();
+    this.sceneManager = sceneManager;
+    this.controls = controls;
+    this.camera = this.createOrthographicCamera();
+    this.GUIFolder = GUIManager.addFolder("HUD");
+
+    this.navigator = new Navigator(this.scene, this.controls, this.GUIFolder);
+    const navigatorParams = { col: 269, row: 143 };
+    //this.navigator.setNavigationTarget(navigatorParams.col, navigatorParams.row);
```
Comment (edisontim) [inline client/src/three/scenes/HUDScene.ts:28]: remove comments

Comment (r0man1337) [reply client/src/three/scenes/HUDScene.ts]: done

File: client/src/three/scenes/HUDScene.ts (hunk 1 above)
Comment (r0man1337) [reply client/src/three/scenes/HUDScene.ts]: done

**Review by cwastche — APPROVED** [review]

**Pull request merged**